dll/win32/shell32/shelldesktop/CChangeNotify.cpp

```diff
@@ -0,0 +1,581 @@
+                                          NULL, (LONG_PTR)pWorker);
+    if (hwndOldWorker == NULL)
+    {
+        ERR("hwndOldWorker == NULL\n");
+        LocalFree(pWorker);
+    }
+
+    DWORD pid;
+    GetWindowThreadProcessId(hwndOldWorker, &pid);
+    TRACE("hwndOldWorker: %p, 0x%lx\n", hwndOldWorker, pid);
+    return hwndOldWorker;
+}
+
+EXTERN_C HANDLE
+DoCreateNotifShare(ULONG nRegID, HWND hwnd, UINT wMsg, INT fSources, LONG fEvents,
+                   LONG fRecursive, LPCITEMIDLIST pidl, DWORD dwOwnerPID)
+{
+    DWORD cbPidl = ILGetSize(pidl);
+    DWORD ibPidl = DWORD_ALIGNMENT(sizeof(NOTIFSHARE));
+    DWORD cbSize = ibPidl + cbPidl;
+    HANDLE hShared = SHAllocShared(NULL, cbSize, dwOwnerPID);
+    if (!hShared)
+    {
+        ERR("Out of memory\n");
+        return NULL;
+    }
+
+    LPNOTIFSHARE pShared = (LPNOTIFSHARE)SHLockSharedEx(hShared, dwOwnerPID, TRUE);
+    if (pShared == NULL)
+    {
+        ERR("SHLockSharedEx failed\n");
+        SHFreeShared(hShared, dwOwnerPID);
+        return NULL;
+    }
+    pShared->dwMagic = NOTIFSHARE_MAGIC;
+    pShared->cbSize = cbSize;
+    pShared->nRegID = nRegID;
+    pShared->hwnd = hwnd;
+    pShared->uMsg = wMsg;
+    pShared->fSources = fSources;
+    pShared->fEvents = fEvents;
+    pShared->fRecursive = fRecursive;
+    pShared->ibPidl = 0;
+    if (pidl)
+    {
+        pShared->ibPidl = ibPidl;
+        memcpy((LPBYTE)pShared + ibPidl, pidl, cbPidl);
+    }
+    SHUnlockShared(pShared);
+    return hShared;
+}
+
+EXTERN_C HANDLE
+DoCreateDeliTicket(LONG wEventId, UINT uFlags, LPCITEMIDLIST pidl1, LPCITEMIDLIST pidl2,
+                   DWORD dwOwnerPID, DWORD dwTick)
+{
+    LPDELITICKET pTicket;
+    HANDLE hTicket = NULL;
+    DWORD cbPidl1, cbPidl2, ibOffset1, ibOffset2, cbSize;
+
+    ibOffset1 = 0;
+    if (pidl1)
+    {
+        cbPidl1 = ILGetSize(pidl1);
+        ibOffset1 = DWORD_ALIGNMENT(sizeof(DELITICKET));
+    }
+
+    ibOffset2 = 0;
+    if (pidl2)
+    {
+        cbPidl2 = ILGetSize(pidl2);
+        ibOffset2 = DWORD_ALIGNMENT(ibOffset1 + cbPidl1);
+    }
+
+    cbSize = ibOffset2 + cbPidl2;
+    hTicket = SHAllocShared(NULL, cbSize, dwOwnerPID);
+    if (hTicket == NULL)
+    {
+        ERR("Out of memory\n");
+        return NULL;
+    }
+
+    pTicket = (LPDELITICKET)SHLockSharedEx(hTicket, dwOwnerPID, TRUE);
+    if (pTicket == NULL)
+    {
+        ERR("SHLockSharedEx failed\n");
+        SHFreeShared(hTicket, dwOwnerPID);
+        return NULL;
+    }
+    pTicket->dwMagic  = DELITICKET_MAGIC;
+    pTicket->wEventId = wEventId;
+    pTicket->uFlags = uFlags;
+    pTicket->ibOffset1 = ibOffset1;
+    pTicket->ibOffset2 = ibOffset2;
+
+    if (pidl1)
+        memcpy((LPBYTE)pTicket + ibOffset1, pidl1, cbPidl1);
+    if (pidl2)
+        memcpy((LPBYTE)pTicket + ibOffset2, pidl2, cbPidl2);
+
+    SHUnlockShared(pTicket);
+    return hTicket;
+}
+
+EXTERN_C LPHANDBAG
+DoGetHandBagFromTicket(HANDLE hTicket, DWORD dwOwnerPID)
```

Comment:
Handbag?!

Reply:
Can you suggest another name?

Reply:
`s/HandBag/Handbag/`.

Reply:
I'm not sure what a "Handbag" represents. "registration entry" perhaps?

Reply:
`SHChangeNotification_Lock` returns a local memory block, that is a handbag.